handlers/create_user.go

```diff
@@ -0,0 +1,46 @@
+package handlers
+
+import (
+	"encoding/json"
+	"http-go/models"
+	"http-go/storage"
+	"io/ioutil"
+	"net/http"
+)
+
+type adder interface {
+	Add(user *models.User) error
+}
+
+func CreateUser(src storage.Adder) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setupResponse(&w)
```

Comment:
You should add `CORS` with middleware that way you don't really need to bother of calling this `setupResponse` mehod. Something like this:

```go
func CORS(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if r.Method == http.MethodOptions {
			w.WriteHeader(http.StatusOK)
			return
		}

		w.Header().Add("Content-Type", "application/json;charset=UTF-8")
		w.Header().Add("Access-Control-Allow-Origin", "*")
		w.Header().Add("Access-Control-Allow-Headers", "*")
		w.Header().Add("Access-Control-Allow-Methods", strings.Join(methods, ","))
		w.Header().Add("Access-Control-Max-Age", "86400")
		w.Header().Add("Content-Type", "application/json")
		next.ServeHTTP(w, r)
	})
}
```

Then in your main function you can do something like:
```go
srv.Handle("/users/create", middleware.CORS(handlers.CreateUser(store)))
```